File: backend/src/types/station/station.ts
type StatusStation = "REAL" | "DIFERIDO" | "AUTOMATICA";
type StationCategory = 'EAA'| 'PE'|'CP'| 'HLM'|'MAP'|'EHA'| 'EHMA'|'CO'|'EMA'|'PLU'| 'EAMA'| 'HLG';
export type StationType = "M" | "H"; // M = Metereólogica ; H = Hidrólogica
type DataMode = "CONVENTIONAL" | "AUTOMATIC";


// Información básica de la estación
export interface StationBase {
  name: string;
  code: string;
  codeOld: string;
  category: StationCategory;
  latitude: number;
  longitude: number;
  status: StatusStation;
  stationType: StationType;
  department?: string;
  province?: string;
  district?: string;
  altitude?: number;
  dataAvailableSince?: number; 
}

// Información para la búsqueda
export interface StationSearch extends Pick<StationBase, "name" | "code" | "stationType"> {
  type: DataMode;
}

// Estación Meteorológica
export interface MeteorologicalStation extends StationBase {
  stationType: "M";
  mode?: DataMode;
}

// Estación Hidrológica
export interface HydrologicalStation extends StationBase {
  stationType: "H";
  mode?: DataMode;
}

// Estaciones
export type Station = MeteorologicalStation | HydrologicalStation;

export interface DataPeriodBase {
  year: number;
  month: number;
  day: number;
}

File: src/garua/schemas/station.py
from garua.models.station import Station

_STATION_TYPE_LABEL = {"M": "METEOROLOGICA", "H": "HIDROLOGICA"}


def station_response_serializer(
    station: Station, distance_km: float | None = None
) -> dict:
    """Convierte un objeto Station a dict serializable para la API."""
    d = {
        "name": station.name,
        "code": station.code,
        "type": _STATION_TYPE_LABEL.get(station.station_type, station.station_type),
        "category": station.category,
        "status": station.status,
        "latitude": station.latitude,
        "longitude": station.longitude,
        "department": station.department,
        "province": station.province,
        "district": station.district,
        "altitude": station.altitude,
        "data_available_since": station.data_available_since,
    }
    if distance_km is not None:
        d["distance_km"] = round(distance_km, 2)
    return d
